packages/server/lib/routes-e2e.ts

```diff
@@ -0,0 +1,156 @@
+import path from 'path'
+import la from 'lazy-ass'
+import check from 'check-more-types'
+import Debug from 'debug'
+import { Router } from 'express'
+
+import AppData from './util/app_data'
+import CacheBuster from './util/cache_buster'
+import specController from './controllers/spec'
+import reporter from './controllers/reporter'
+import { runner } from './controllers/runner'
+import client from './controllers/client'
+import files from './controllers/files'
+import type { InitializeRoutes } from './routes'
+
+const debug = Debug('cypress:server:routes-e2e')
+
+export const createRoutesE2E = ({
+  config,
+  specsStore,
+  getRemoteState,
+  networkProxy,
+  getSpec,
+  getCurrentBrowser,
+  onError,
+  testingType,
+}: InitializeRoutes) => {
+  const routesE2E = Router()
+
+  // routing for the actual specs which are processed automatically
+  // this could be just a regular .js file or a .coffee file
+  routesE2E.get('/__cypress/tests', (req, res, next) => {
+    // slice out the cache buster
+    const test = CacheBuster.strip(req.query.p)
+
+    specController.handle(test, req, res, config, next, onError)
+  })
+
+  routesE2E.get('/__cypress/socket.io.js', (req, res) => {
+    client.handle(req, res)
+  })
+
+  routesE2E.get('/__cypress/reporter/*', (req, res) => {
+    reporter.handle(req, res)
+  })
+
+  routesE2E.get('/__cypress/automation/getLocalStorage', (req, res) => {
```

Comment:
I hope that as we unify the runners, the amount of `e2e` specific routes will be less, moving to `routes.ts`. For example, this route probably isn't supposed to be e2e only - CT should have this local storage automation feature, too.

This refactor makes it more clear what should and should not be shared, and can be improved as we move forward.